app/src/main/java/com/islamey/hamza/wazaker/ui/CounterFragment/CounterFragment.kt

```diff
@@ -1,86 +1,68 @@
 
-    private val MY_PREFS_NAME = "pref"
-    private var sharedPreferences: SharedPreferences? = null
-
-    private var counts = 0
-    private var totalCounts = 0
+    private val counterViewModel: CounterViewModel by viewModels()
 
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View {
+        setHasOptionsMenu(true)
+        binding = FragmentCounterBinding.inflate(layoutInflater, container, false)
+        return binding.root
+    }
 
-        _binding = FragmentCounterBinding.inflate(layoutInflater, container, false)
-
-        getDataFromSharedPreferences()
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
 
-        binding.counterscreen.setOnClickListener {
-            counts++
-            totalCounts++
+        initUi()
+    }
 
-            checkCounts(counts)
-            binding.countertextview.text = counts.toString()
-            setTotalZekerCountstext(totalCounts)
+    @SuppressLint("SetTextI18n")
+    private fun initUi() {
 
-            saveTotalcountsInSharedPreference()
+        lifecycleScope.launch {
+            counterViewModel.counts.collect { counts ->
+                binding.countertextview.text = counts.toString()
+                checkCounts(counts)
+            }
         }
 
-        binding.resetbtn.setOnClickListener {
-            counts = 0
-            binding.countertextview.text = counts.toString()
-            binding.countertextview.setTextColor(Color.WHITE)
+        lifecycleScope.launch {
+            counterViewModel.totalCounts.collect { totalCounts ->
+                binding.totalCountstextview.text =
+                    getString(R.string.totalzeker) + "  " + totalCounts.toString()
+            }
         }
 
+        binding.counterScreenLayout.setOnClickListener {
+            counterViewModel.incrementCounter()
+        }
 
-        return binding.root
-    }
-
-    @SuppressLint("SetTextI18n")
-    fun getDataFromSharedPreferences() {
-        sharedPreferences = context?.getSharedPreferences(MY_PREFS_NAME, Context.MODE_PRIVATE)
-        totalCounts = sharedPreferences!!.getInt("zekertotalcounts", totalCounts)
-        setTotalZekerCountstext(totalCounts)
-    }
-
-    @SuppressLint("SetTextI18n")
-    fun setTotalZekerCountstext(Totalzekercounts: Int) {
-        binding.totalCountstextview.text = getString(R.string.totalzeker) + "  " + Totalzekercounts
-    }
-
-    private fun checkCounts(counts: Int) {
-        if (counts % 33 == 0) getRandomcolor()
-    }
-
-    private fun getRandomcolor() {
-        val currentColor: Int
-        if (counts % 33 == 0) {
-            currentColor = RandomColor.getRandomColor()
-            binding.countertextview.setTextColor(currentColor)
+        binding.resetbtn.setOnClickListener {
+            counterViewModel.resetCounter()
```

Comment:
as we have `binding` we can set it from binding itself by passing a reference to view model to data binding